### DISCUSSION TITLE
Rescript-Remix and useFetcher

### DISCUSSION THREAD

**Discussion by youngkidwarrior** [remix-run/remix#3692] [Q&A] [answered]
Im trying to make a resource route from an unconventional route filename. 

when I call `fetcher.load()` I am getting an error where the data is looking in the unconventional route

my folder structure looks like this
```
app/
├── res-routes/
│   └── Root_FetchGuilds.js
│   └── Root_Index.js
├── routes/
│   └── index.js
│   └── fetchGuilds.js
└── root.res
```

`routes/fetchGuilds.js` imports from `res-routes/Root_FetchGuilds` 

```
import { loader } from '~/res-routes/Root_FetchGuilds.js'

export { loader }
```

I am getting this error:

```Error: Route "res-routes/Root_FetchGuilds" does not match URL "/fetchGuilds"```

and the URL from useFetcher looks like this

```/fetchGuilds?_data=res-routes/Root_FetchGuilds```

when it should be 
```/fetchGuilds?_data=routes/fetchGuilds```

**Comment by kiliman** [comment, marked as answer]
Remix will only generate routes for files in the `app/routes` folder. If you need to add additional routes, you'll need to use *remix.config* to setup your routes manually.

https://remix.run/docs/en/v1/api/conventions#routes

**Reply by youngkidwarrior** [reply]
This is a script written for rescript-remix to handle route creation

https://github.com/tom-sherman/rescript-remix/blob/main/routing/RouteConventions.res
